### DISCUSSION TITLE
Emoji trouble [??? Instead of emoji] Directus v9.7

### DISCUSSION THREAD

**Discussion by 53rg0** [directus/directus#12287] [Feature Requests]
When I try to add emoji, I got in panel ??? Instead of 🚀.

I carefully read all the solutions that were previously suggested and nothing helps.
I tried to play with DB_CHARSET="utf8mb4"
I tried to play with DB_CONNECTION_STRING="mysql://u:p@127.0.0.1/d?debug=false&charset=UTF8MB4"

(Of course I have tried everything with utf8mb4_unicode_ci, utf8mb4_general_ci etc)

Nothing helped. My database 100% utf8mb4, when I connect with client to my database and use Client Character Set "utf8mb4" encoding "utf-8" and my database, table, column are utf8mb4 utf8mb4_unicode_ci I see everything fine.

When I add emoji in database directly in panel, I got only one ? in panel

I've been fighting for these emojis all day and going crazy. 
Help)

### To Reproduce

Just try to add emoji in panel

### What version of Directus are you using?

9.7

### What database are you using?

MariaDB 10.5.15

**Comment by rijkvanzanten** [comment]
> My database 100% utf8mb4

Are you 100,000% sure your database and all tables/columns are utf8mb4? If emoji's are saved as `???`, that is most definitely caused by the wrong collation in mysql/mariadb (Directus itself doesn't do any magic utf parsing)

**Reply by 53rg0** [reply]
![Снимок экрана 2022-03-21 в 22 12 33](https://user-images.githubusercontent.com/40628373/159355918-41f6390f-e964-4ff5-9d4e-9461a76466fe.png)

100%

and I try to set DB_CHARSET="utf8mb4"

does not work...

**Reply by rijkvanzanten** [reply]
And — yes this is an obvious question, but — your directus instance is connected to this exact database? It has to be a collation/charset problem in the database connection or database itself, as that's the only point at which emoji's might be converted to `?` / `???`

**Reply by 53rg0** [reply]
100% directus instance is connected to exact database
I think 99% problem is in "client charset", looks like my DB_CHARSET settings does not work

**Reply by 53rg0** [reply]
Problem with read\write from directus, so this is connection problem

DB_CONNECTION_STRING="mysql://u:pass@127.0.0.1/backendd?debug=false&charset=utf8mb4" also does not work

**Reply by rijkvanzanten** [reply]
Have you tried using the individual env vars with the charset? So instead of the connection string, use `DB_HOST` `DB_USER` etc etc 🤔

**Comment by 53rg0** [comment]
I connected in many ways, in general everything works, except for emojis
✨📲⚡️❄️☺️🔍⚙️ this is how this line is saved
✨????⚡️❄️☺️????⚙️

Another example
🌈⛈🎉🌹🐧😊
????⛈????????????????
But! When i add emoji directly in database it looks good
![Снимок экрана 2022-03-22 в 08 16 27](https://user-images.githubusercontent.com/40628373/159419435-f7eb5c9f-28d1-42de-b16f-9d9c35758dc1.png)
But in directus panel i see now ? per emoji
![Снимок экрана 2022-03-22 в 08 17 25](https://user-images.githubusercontent.com/40628373/159419565-b325b06c-cf58-4f04-8816-e6806d2abba8.png)

As you can see, some emojis have been saved normally

SO! It's 100% connection problem, not in database
From all emojis works only ~20

I think DB_CHARSET does not work

**Comment by 53rg0** [comment]
DB_CHARSET="utf8mb4" does not work
DB_CHARSET="utf8mb4_unicode_ci" does not work

**My .env**

DB_CLIENT="mysql"
DB_HOST="127.0.0.1"
DB_PORT="3306"
DB_DATABASE="backendd"
DB_USER="u_backendd"
DB_PASSWORD="pass"
DB_CHARSET="utf8mb4_unicode_ci" - any actions with this parameter does not affect
#DB_CONNECTION_STRING="mysql://u_backendd:pass@127.0.0.1/backendd?debug=false&charset=utf8mb4"

**Comment by 53rg0** [comment]
I tried to reinstall on different hosting - it did not help, there is a problem

**Comment by rijkvanzanten** [comment]
Using a totally fresh container from `mysql:8`, with a new database with the default collation (which is apparently `utf8mb4_0900_ai_ci`, according to `SELECT @@collation_database;`), installing Directus fresh using bootstrap (without any `DB_CHARSET` configured), you end up with issues inserting some of the more ""modern"" emoji (for example ⚡️ works, 🌈 throws an error: `ER_TRUNCATED_WRONG_VALUE_FOR_FIELD`). 

When setting `DB_CHARSET="utf8mb4"`, and installing Directus, we have more luck:

<img width="729" alt="CleanShot 2022-03-22 at 11 18 02@2x" src="https://user-images.githubusercontent.com/9141017/159516078-dbd846e1-40ac-4fb9-be3d-edaf7929f8d3.png">

**Comment by ludi81** [comment]
If I add the following line to the .env file: DB_CHARSET="utf8mb4_unicode_ci" it works fine. 

But I receive the following warning on startup:
Using custom DB_CHARSET "utf8mb4_unicode_ci". Using a charset different from the database's default can cause problems in relationships. Omitting DB_CHARSET is strongly recommended.

I changed the database default charset to utf8mb4_unicode_ci and all tables and columns to utf8mb4_unicode_ci. I still receive the above warning. If I remove the  DB_CHARSET="utf8mb4_unicode_ci" line, the emoji doesn't work any more.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
